test/unit_test/api/apps/services/test_provider_api_service_verify_discovery.py

```
@@ -0,0 +1,238 @@
+    logged = caplog.text
+    for secret in ("s3cr3t-token", "AKIA-SECRET", "svc:"):
+        assert secret not in message, message
+        assert secret not in logged, logged
+    assert "https://***@models.internal:8443/v1" in message
+
+
+@pytest.mark.asyncio
+async def test_authority_less_base_url_is_not_echoed_back(monkeypatch):
+    """`urlparse` happily accepts a URL with no authority, e.g. one the user typed
+    without a scheme. There is no host to keep, so nothing may be echoed."""
+    secret_url = "models.internal/v1?api-key=AKIA-SECRET"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "secret_url"

(S105)